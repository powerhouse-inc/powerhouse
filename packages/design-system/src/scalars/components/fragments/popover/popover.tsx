import * as PopoverPrimitive from "@radix-ui/react-popover";
import React from "react";
import { cn } from "../../../lib/utils.js";

export const Popover = PopoverPrimitive.Root;
export const PopoverAnchor = PopoverPrimitive.Anchor;
export const PopoverTrigger = PopoverPrimitive.Trigger;

export const PopoverContent = React.forwardRef<
  React.ElementRef<typeof PopoverPrimitive.Content>,
  React.ComponentPropsWithoutRef<typeof PopoverPrimitive.Content>
>(({ align = "center", sideOffset = 4, className, ...props }, ref) => (
  <PopoverPrimitive.Portal>
    <PopoverPrimitive.Content
      align={align}
      sideOffset={sideOffset}
      className={cn(
        [
          "data-[state=closed]:animate-fade-out data-[state=open]:animate-fade-in",
          "data-[state=closed]:animate-zoom-out data-[state=open]:animate-zoom-in",
          "data-[side=bottom]:animate-slide-in-from-top data-[side=left]:animate-slide-in-from-right",
          "data-[side=right]:animate-slide-in-from-left data-[side=top]:animate-slide-in-from-bottom",
          "w-(--radix-popover-trigger-width) z-50 border p-0 outline-none",
          "border-gray-300 bg-white dark:border-slate-500 dark:bg-slate-600",
          "rounded-md shadow-[1px_4px_15px_0px_rgba(74,88,115,0.25)] dark:shadow-[1px_4px_15.3px_0px_#141921]",
        ],
        className,
      )}
      ref={ref}
      {...props}
    />
  </PopoverPrimitive.Portal>
));
