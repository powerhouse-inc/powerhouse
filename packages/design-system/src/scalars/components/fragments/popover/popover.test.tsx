import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { Popover } from "./popover.js";

describe("Popover Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(<Popover />);
    expect(asFragment()).toMatchSnapshot();
  });
});
