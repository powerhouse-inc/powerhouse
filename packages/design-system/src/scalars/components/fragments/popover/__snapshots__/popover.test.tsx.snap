// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Popover Component > should match snapshot 1`] = `<DocumentFragment />`;
